tests/TypeShape.Tests/TestTypes.cs

```diff
@@ -592,15 +592,27 @@ public StructWithDefaultCtor()
 }
 
 [GenerateShape]
-public partial class BaseClass
+public partial class BaseClass : IEquatable<BaseClass>
```

Comment:
Would it make sense to factor the IEquatable implementation to a separate set of types so that we coverage for both cases?

Reply:
I added this because I had been testing round-tripping. I'm not sure what case we're *not* covering when this interface isn't present. 
But I _did_ just discover `IsEquatable` on test cases, which probably was `false` for this type up to this point and why no one else hit the fact that this class didn't know how to do a by-value compare of itself. 
I can remove this change if you'd like. I don't need it any more.